libexec/bats-core/bats-exec-test

```diff
@@ -40,24 +40,21 @@ fi
 BATS_TEST_DIRNAME="${BATS_TEST_FILENAME%/*}"
 BATS_TEST_NAMES=()
 
+# Shorthand for source-ing files relative to the BATS_TEST_DIRNAME,
+# optionally with a .bash suffix appended. If the argument doesn't
+# resolve relative to BATS_TEST_DIRNAME it is sourced as-is.
 load() {
-  local name="$1"
-  local filename
-
-  if [[ "${name:0:1}" == '/' ]]; then
-    filename="${name}"
-  else
-    filename="$BATS_TEST_DIRNAME/${name}.bash"
-  fi
+  local file="${1:?name}"
 
-  if [[ ! -f "$filename" ]]; then
-    printf 'bats: %s does not exist\n' "$filename" >&2
-    exit 1
+  if [[ -f "${BATS_TEST_DIRNAME}/${file}" ]]; then
+    file="${BATS_TEST_DIRNAME}/${file}"
+  elif [[ -f "${BATS_TEST_DIRNAME}/${file}.bash" ]]; then
+    file="${BATS_TEST_DIRNAME}/${file}.bash"
   fi
 
-  # Dynamically loaded user files provided outside of Bats.
+  # Dynamically loaded user file provided outside of BATS.
   # shellcheck disable=SC1090
-  source "${filename}"
+  source "${file}" || exit
```

Comment:
What was the previous behaviour here, silent fail? If so perhaps this could be something like

```bash
if ! source "${file}"; then
  printf error message
fi
```

Reply:
It would return a non-zero exit code (and print an error), but allow the script to proceed. I imagine the prior assumption was that sourcing would always succeed if the file existed, which isn't necessarily true (e.g. lacking read permission or syntax errors in the file).

That said, this points out another behavior change which may be important. `source` will also return a non-zero exit code if the sourced script's final command does so. If that's a problem we can't actually rely on `source` for error-handling after all (though I doubt users _intend_ for sourced scripts to have non-zero exit codes).